server/client.go

```diff
@@ -441,7 +440,7 @@ func (c *client) initClient() {
 
 	// Outbound data structure setup
 	c.out.sz = startBufSize
-	c.out.sg = sync.NewCond(&c.mu)
+	c.out.sch = make(chan struct{}, 1)
```

Comment:
If we do select for sending can be non-buffered I think.

Reply:
No, because if the writeLoop is not receiving on this channel then the signal code will fall into default (since it can't send) and that will mean missing a notification.

Reply:
And that is why I love working with you. ;)